File: APP/src/features/search/responses/search-engine-graph-data.response.ts
import { SearchEnginePlotDataResponse } from '@/features/search/responses/search-engine-plot-data.response.ts';
import { SearchEngine } from '@/common/enums/SearchEngine.ts';

export interface SearchEngineGraphDataResponse {
  searchEnginePlotData: Record<SearchEngine, SearchEnginePlotDataResponse>;
}

File: API/InfoTrack.Domain/Responses/SearchEngineGraphDataResponse.cs
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using InfoTrack.Domain.Enums;

namespace InfoTrack.Domain.Responses;

[ExcludeFromCodeCoverage]
public record SearchEngineGraphDataResponse
{
    public Dictionary<SearchEngine, SearchEnginePlotDataResponse> SearchEnginePlotData { get; init; }

    public SearchEngineGraphDataResponse(Dictionary<SearchEngine, SearchEnginePlotDataResponse> searchEnginePlotData)
    {
        SearchEnginePlotData = searchEnginePlotData;
    }
}